### PR TITLE
Add filters & sorting for pnp extraction errors on Progress Reports

### PR DESCRIPTION
**Pull request by rdonigian**
Adds filter
```ts
ProgressReportFilters.pnpExtractionResult.hasError
```
and sorting
```ts
ProgressReport.pnpExtractionResult.totalErrors
```

[Monday](https://seed-company-squad.monday.com/boards/5989610236/views/131962501/pulses/7893107215)

## Summary by CodeRabbit

- **New Features**
	- Introduced filtering capabilities for extraction results with a new filter class.
	- Added optional error tracking in extraction results.
	- Enhanced progress report filtering to include conditions based on extraction results.
	- Added sorting functionality for extraction results based on error counts.

- **Bug Fixes**
	- Improved error handling in extraction result queries.

- **Documentation**
	- Updated documentation to reflect new filtering options and error tracking features.

### CODE REVIEW

**Review by atGit2021 — COMMENTED** [review]
LGTM!

File: src/components/progress-report/progress-report.repository.ts [hunk 1]
```diff
@@ -109,5 +110,14 @@ export const progressReportFilters = filter.define(
         node('node', 'Engagement'),
       ]),
     ),
+    pnpExtractionResult: filter.sub(() => pnpExtractionResultFilters)((sub) =>
+      sub.match([
+        node('outer'),
+        relation('out', '', 'reportFileNode'),
+        node('file', 'File'),
+        relation('out', '', 'pnpExtractionResult'),
+        node('result', 'PnpExtractionResult'),
```
Comment (CarsonF) [inline src/components/progress-report/progress-report.repository.ts:119]:
```suggestion
        node('node', 'PnpExtractionResult'),
```

File: src/components/pnp/extraction-result/extraction-result.dto.ts [hunk 2]
```diff
@@ -71,6 +86,9 @@ export abstract class PnpExtractionResult {
     });
   }
 }
+
+export { PnpExtractionResult as IPnpExtractionResult };
+
```
Comment (CarsonF) [inline src/components/pnp/extraction-result/extraction-result.dto.ts:91]: Not needed
```suggestion
```

File: src/components/pnp/extraction-result/pnp-extraction-result.neo4j.repository.ts [hunk 3]
```diff
@@ -71,3 +79,14 @@ export class PnpExtractionResultNeo4jRepository
...
+
+export const pnpExtractionResultFilters = filter.define(
+  () => PnpExtractionResultFilters,
+  {
+    hasError: ({ value, query }) => query.where({ 'result.hasError': value }),
+  },
+);
+
+export const pnpExtractionResultSorters = defineSorters(IPnpExtractionResult, {
+  countError: (query) => query.return<SortCol>('node.countError as sortValue'),
```
Comment (CarsonF) [inline src/components/pnp/extraction-result/pnp-extraction-result.neo4j.repository.ts:91]: Can avoid this by

```ts
class PnpExtractionResult {
  static readonly BaseNodeProps = ['countError'];
}
```

File: src/components/pnp/extraction-result/pnp-extraction-result.neo4j.repository.ts [hunk 4]
```diff
@@ -71,3 +79,14 @@ export class PnpExtractionResultNeo4jRepository
       .executeAndLogStats();
   }
 }
+
+export const pnpExtractionResultFilters = filter.define(
+  () => PnpExtractionResultFilters,
+  {
+    hasError: ({ value, query }) => query.where({ 'result.hasError': value }),
```
Comment (CarsonF) [inline src/components/pnp/extraction-result/pnp-extraction-result.neo4j.repository.ts:86]:
```suggestion
    hasError: ({ value, query }) => query.where({ 'node.hasError': value }),
```

File: src/components/pnp/extraction-result/extraction-result.dto.ts [hunk 5]
```diff
@@ -40,9 +41,23 @@ export class PnpProblem {
   readonly groups: readonly string[];
 }
 
+@InputType()
+export class PnpExtractionResultFilters {
+  @Field(() => Boolean, {
+    nullable: true,
+    description: 'Filter by hasError',
```
Comment (CarsonF) [inline src/components/pnp/extraction-result/extraction-result.dto.ts:48]: Doesn't add any clarity
```suggestion
```

**Review by CarsonF — CHANGES_REQUESTED** [review]
The hasError filter vs countErrors sort is kinda weird.
Maybe we could just call both `errors`?
Idk maybe that's ambiguous and limits future filters.

**Comment by rdonigian** [comment]
> The hasError filter vs countErrors sort is kinda weird. Maybe we could just call both `errors`? Idk maybe that's ambiguous and limits future filters.

I was thinking about this but wasn't sure about how to navigate bc they're not exactly the same thing and to your point could put us in a corner in the future.

**Review by CarsonF — APPROVED** [review]

**Pull request merged**